kv/dist_sender.go

```diff
@@ -647,42 +671,71 @@ func (ds *DistSender) Send(_ context.Context, call client.Call) {
 			call.Reply.Header().SetGoError(err)
 			return
 		}
+
+		if finalReply != curReply {
+			// This was the second or later call in a multi-range request.
+			// Combine the new response with the existing one.
+			_, ok := finalReply.(proto.Combinable)
+			if !ok {
```

Comment:
That's what I had originally, but thought it looked weird to have

```
cFinalReply, ok := ...
if !ok { ... }
cFinalReply.Combine(curReply)
```

because it obfuscates that it alters `finalReply`.